src/util/symbol_table_base.h

```diff
@@ -117,6 +117,78 @@ class symbol_table_baset
   virtual void clear() = 0;
 
   void show(std::ostream &out) const;
+
+  class iteratort
+  {
+  private:
+    symbolst::iterator it;
+    std::function<void(const irep_idt &id)> on_update;
```

Comment:
Rename `on_get_writeable` -- the fact that it's actually `journalling_symbol_tablet::on_update` is none of this class' business ;)